node_modules/gh-orgunit/tests/test-import.js

```diff
@@ -0,0 +1,806 @@
+                                                OrgUnitTestsUtil.assertImportOrgUnitImportFails(global.tests.admins.cam2013.client, part.id, importData, false, ['Nesting a part under a module is invalid'], function() {
+                                                    importData = _generateStructure('course', 'part', 'module', 'subject');
+                                                    OrgUnitTestsUtil.assertImportOrgUnitImportFails(global.tests.admins.cam2013.client, part.id, importData, false, ['Nesting a subject under a module is invalid'], function() {
+                                                        importData = _generateStructure('course', 'part', 'module', 'course');
+                                                        OrgUnitTestsUtil.assertImportOrgUnitImportFails(global.tests.admins.cam2013.client, part.id, importData, false, ['Nesting a course under a module is invalid'], function() {
+                                                            importData = _generateStructure('course', 'part', 'module', 'module');
+                                                            OrgUnitTestsUtil.assertImportOrgUnitImportFails(global.tests.admins.cam2013.client, part.id, importData, false, ['Nesting a module under a module is invalid'], function() {
+                                                                return callback();
+                                                            });
+                                                        });
+                                                    });
+                                                });
+                                            });
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Test that verifies that organisational units get validated
+         */
+        it('verify basic organisational unit validation', function(callback) {
+            OrgUnitTestsUtil.assertCreateOrgUnit(global.tests.admins.cam2013.client, 'Test part', 'part', {}, function(part) {
+
+                // Too long displayName
+                var importData = _getTestTree('1part-2module-1series-1event');
+                importData.children[0].displayName = TestsUtil.generateString(257);
+                OrgUnitTestsUtil.assertImportOrgUnitImportFails(global.tests.admins.cam2013.client, part.id, importData, false, ['displayName'], function() {
+
+                    // Too long type
+                    importData = _getTestTree('1part-2module-1series-1event');
+                    importData.children[0].type = TestsUtil.generateString(33);
+                    OrgUnitTestsUtil.assertImportOrgUnitImportFails(global.tests.admins.cam2013.client, part.id, importData, false, ['type'], function() {
```

Comment:
Aren't there an enumerated set of types we can validate against? Or can consumers effectively make their own types?